index.html

```diff
@@ -1127,73 +1676,262 @@ <h2>
+                <p>
+                  Let |fallbackTimestamp| be [=current coarsened wall time=].
+                </p>
+              </li>
+              <li>
+                <p>
+                  Let |declarativeResult| be the result of running the [=/declarative push message
+                  parser=] given |bytes|, |origin|, |baseURL|, and |fallbackTimestamp|.
+                </p>
+              </li>
+              <li>
+                <p>
+                  If |declarativeResult| is not failure:
+                </p>
                 <ol>
-                  <li>Acknowledge the receipt of the <a>push message</a> according to [[RFC8030]].
-                  Though the message was not successfully received and processed, this prevents the
-                  push service from attempting to retransmit the message; a badly encrypted message
-                  is not recoverable.
+                  <li>
+                    <p>
+                      Let |notification| be |declarativeResult|'s [=declarative push message parser
+                      result/notification=].
+                    </p>
+                  </li>
+                  <li>
+                    <p>
+                      Set |notification|'s [=notification/service worker registration=] to
+                      |registration|.
+                    </p>
                   </li>
-                  <li>Abort these steps.
+                  <li>
+                    <p>
+                      Let |notificationShown| be false.
+                    </p>
+                  </li>
+                  <li>
+                    <p>
+                      Let |appBadgeSet| be false.
+                    </p>
+                  </li>
+                  <li>
+                    <p>
+                      If |declarativeResult|'s [=declarative push message parser result/mutable=]
+                      is true:
+                    </p>
+                    <ol>
+                      <li>
+                        <p>
+                          Let |result| be the result of [=fire a push event|firing a push event=]
+                          given |registration|, null, a new {{Notification}} object representing
+                          |notification|, and |declarativeResult|'s [=declarative push message
+                          parser result/app badge=].
+                        </p>
+                      </li>
+                      <li>
+                        <p>
+                          If |result| is not failure, then set |notificationShown| to |result|'s
+                          [=push event result/notification shown=] and |appBadgeSet| to |result|'s
+                          [=push event result/app badge set=].
+                        </p>
+                      </li>
+                    </ol>
+                  </li>
+                  <li>
+                    <p>
+                      If |notificationShown| is false, then run the [=notification show steps=]
```

Comment:
Noticed during an internal discussion; this would involve cookies as current implemented in Firefox, and clearing site data would mean the cookie jar would become empty, which may be significant if the image URL can only be fetched with proper cookie (imagine an image attachment from a DM). Should we accept such breakage? It seems like that should be somehow covered in the spec at least. (related: https://github.com/whatwg/notifications/issues/5)

Reply:
It's not entirely clear to me what you are suggesting here.

Reply:
Hmm. Isn't that just an artifact from how cookies work as well? Should every specification that potentially fetches something later also point out that cookies currently expire after a little over a year?

Reply:
(But good point, it's not really DWP specific but also affects the existing Push)